pkg/controllers/management/auth/auth_config.go

```diff
@@ -33,29 +34,46 @@ type CleanupService interface {
 }
 
 type authConfigController struct {
-	users            v3.UserLister
-	authRefresher    providerrefresh.UserAuthRefresher
-	cleanup          CleanupService
-	authConfigClient controllers.AuthConfigClient
+	users         v3.UserLister
+	authRefresher providerrefresh.UserAuthRefresher
+	cleanup       CleanupService
+	authConfigs   objectclient.GenericClient
```

Comment:
Nit:

Generally not a fan of the name here. Might be good to call this something like "authConfigsUnstructured".

More importantly, there's also no comment here explaining why we need to use an unstructured client for authconfigs instead of the fully-typed authconfig interface. Given that this has already bit us once, I think adding a comment explaining the whole "auth configs have fields that are internal to types and so we need to use the unstructured client to update" situation is a good idea.